tools/schemas/enclave-config.schema.json

```diff
@@ -0,0 +1,539 @@
+        "string",
+        "number"
+      ],
+      "pattern": "^0$|^[1-9][0-9]*$",
+      "maxLength": 10,
+      "minimum": 0,
+      "maximum": 4294967296,
+      "multipleOf": 1.0
+    },
+    "safe_uint64_t": {
+      "type": [
+        "string",
+        "number"
+      ],
+      "pattern": "^0$|^[1-9][0-9]*$",
+      "maxLength": 20,
+      "minimum": 0,
+      "maximum": 9007199254740992,
+      "multipleOf": 1.0
+    },
+    "safe_size_t": {
+      "type": [
+        "string",
+        "number"
+      ],
+      "pattern": "^0$|^[1-9][0-9]*$",
+      "maxLength": 20,
+      "minimum": 0,
+      "maximum": 9007199254740992,
+      "multipleOf": 1.0
+    },
+    "sgxlkl_enclave_mode_t": {
+      "enum": [
+        "unknown",
+        "sw_debug",
+        "hw_debug",
+        "hw_release"
+      ],
+      "c_enum": [
+        "UNKNOWN_MODE",
+        "SW_DEBUG_MODE",
+        "HW_DEBUG_MODE",
+        "HW_RELEASE_MODE"
+      ]
+    },
+    "sgxlkl_enclave_mmap_files_t": {
+      "enum": [
+        "none",
+        "private",
+        "shared"
+      ],
+      "c_enum": [
+        "ENCLAVE_MMAP_FILES_NONE",
+        "ENCLAVE_MMAP_FILES_PRIVATE",
+        "ENCLAVE_MMAP_FILES_SHARED"
+      ]
+    },
+    "sgxlkl_exit_status_mode_t": {
+      "enum": [
+        "full",
+        "binary",
+        "none"
+      ],
+      "c_enum": [
+        "EXIT_STATUS_FULL",
+        "EXIT_STATUS_BINARY",
+        "EXIT_STATUS_NONE"
+      ]
+    },
+    "Elf64_auxv_t": {
+      "type": "object",
+      "properties": {
+        "a_type": {
+          "type": "string"
+        },
+        "a_val": {
+          "type": "string"
+        }
+      }
+    },
+    "sgxlkl_clock_res_config_t": {
+      "type": "object",
+      "properties": {
+        "resolution": {
+          "type": "string",
+          "minLength": 16,
+          "maxLength": 16,
+          "pattern": "^[0-9A-Fa-f]+$",
+          "default": "0000000000000000"
+        }
+      }
+    },
+    "sgxlkl_enclave_mount_config_t": {
+      "type": "object",
+      "description": "Disk configuration",
+      "required": [
+        "destination"
+      ],
+      "additionalProperties": false,
+      "dependencies": {
+        "roothash": [
+          "roothash_offset"
+        ],
+        "roothash_offset": [
+          "roothash"
+        ],
+        "create": [
+          "size"
+        ],
+        "size": [
+          "create"
+        ]
+      },
+      "properties": {
+        "create": {
+          "type": "boolean",
+          "description": "Whether to dynamically create the disk (note that an empty image file must exist).",
+          "default": false
+        },
+        "destination": {
+          "type": "string",
+          "description": "Mount point",
+          "pattern": "^/.*$",
+          "maxLength": 255
+        },
+        "key": {
+          "type": [
+            "array",
+            "null"
+          ],
+          "description": "Disk encryption key (hex-encoded).",
```

Comment:
A "simple string"? All the tests pass, so no, it doesn't break anything. There may be a couple pass phrases here or there, but the launcher parses those independently before they get re-encoded into an enclave config.

Reply:
See for example tests/containers/encrypted/app-config-integrity/Makefile. The key is a string, not an array. Keep in mind that some tests especially related to disk encryption are disabled in CI.

Reply:
In `sgxlkl_enclave_config_t`, keys are `uint8_t*`, not `char*` and the code relies on that (new and old, even though it was previously at some point labelled a `char*` I think). If there are tests that assume otherwise, we'll have to fix them.

Reply:
The key is generated by sgx-lkl-disk, it's a hex-encoded string. This string can be copied into the "key" field. This is broken now. Do you see the issue?

Reply:
No. If it was every copied into the key field (in the struct), that was a bug, because the rest of the code assumes that it isn't. That's why there is a `key_len` and no null-termination.

Reply:
Did you have a look at the test I mentioned above? Maybe that clears things up.

Reply:
That tests uses a (binary) key file, but it translates it into hex before inserting it into the config (`$(shell xxd ...`). I just had to adapt the config template to the new structure, now the test passes.

Reply:
I dimly remember that somewhere there used to be a test that used "123ABC" I think, but conveniently that's actually legal hex.

Reply:
Do you mean https://github.com/lsds/sgx-lkl/pull/290/commits/70cfa568337318e434c3d7aed9a2b9207d11b774? It still uses a string. How does that work if the type is an array?

Reply:
Oh, now I get what you mean. In the schema it's declared as an array of `uint8_t` so that the right C struct is extracted. Yeah, I'll add a new type for that, it should be a hex-string in the schema.

Reply:
Done.